openedx/core/djangoapps/django_comment_common/utils.py

```diff
@@ -134,34 +134,3 @@ def get_course_discussion_settings(course_key):
         )
 
     return course_discussion_settings
-
-
-def set_course_discussion_settings(course_key, **kwargs):
-    """
-    Set discussion settings for a course.
-
-    Arguments:
-        course_key: CourseKey
-        always_divide_inline_discussions (bool): If inline discussions should always be divided.
-        divided_discussions (list): List of discussion ids.
-        division_scheme (str): `CourseDiscussionSettings.NONE`, `CourseDiscussionSettings.COHORT`,
-            or `CourseDiscussionSettings.ENROLLMENT_TRACK`
-
-    Returns:
-        A CourseDiscussionSettings object.
-    """
-    fields = {
-        'division_scheme': (str,)[0],
-        'always_divide_inline_discussions': bool,
-        'divided_discussions': list,
-    }
-
-    course_discussion_settings = get_course_discussion_settings(course_key)
```

Comment:
We were always doing a `get` for the settings during an `update`.
In practice, nearly everywhere we were calling `update`, we had already done a `get` to lookup the object. By converting this helper _function_ into a helper _method_ on the object, we can avoid doing the second lookup.
In the event code paths do _not_ already have an instance, we can just `get` it and `.update` it, which is still at worst one lookup.